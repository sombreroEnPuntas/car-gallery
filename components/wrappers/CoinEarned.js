import styled from 'styled-components'
import { animations } from '.'
import Coin from './Coin'

const CoinEarned = styled(Coin)`
  animation: ${animations.rotateIn} 1.4s ease-out forwards,
    ${animations.fadeOut} 2s ease-out forwards;

  position: absolute;
`

export default CoinEarned
