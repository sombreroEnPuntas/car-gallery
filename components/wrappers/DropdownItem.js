import styled from 'styled-components'
import { fonts } from '.'

export const DropdownItem = styled.li`
  align-items: center;
  display: flex;
  height: ${fonts.lineHeight};
  justify-content: center;
  margin: 0;
  overflow: hidden;
  padding: 0;
  text-align: center;
  text-decoration: none;
  text-overflow: ellipsis;
  text-transform: capitalize;
  white-space: nowrap;
  width: 100%;
`

export default DropdownItem
