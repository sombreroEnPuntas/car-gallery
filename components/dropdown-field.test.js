// @flow
import React from 'react'
import { shallow } from 'enzyme'

// Dependencies
import CoinEarned from '../components/wrappers/CoinEarned'
import DropdownItem from '../components/wrappers/DropdownItem'
import Input from '../components/wrappers/Input'

// Tested unit
import TestedComponent from './dropdown-filed'

const handleUpdate = jest.fn()
const getProps = customProps => ({
  autocomplete: 'dinosaur-name',
  disabled: false,
  handleUpdate,
  name: 'dinosaur-name',
  valid: false,
  value: '',
  values: ['t-rex', 'brontosaurus'],
  ...customProps,
})

describe('DropdownField', () => {
  describe.each`
    status              | Component       | event                          | options | value
    ${'char typed'}     | ${Input}        | ${{ target: { value: 'r' } }}  | ${2}    | ${'r'}
    ${'char typed'}     | ${Input}        | ${{ target: { value: 'x' } }}  | ${1}    | ${'x'}
    ${'option clicked'} | ${DropdownItem} | ${{ target: { id: 't-rex' } }} | ${1}    | ${'t-rex'}
  `('$status', ({ Component, event, options, value }) => {
    const wrapper = shallow(<TestedComponent {...getProps({ value })} />)

    it(`handles the event properly`, () => {
      wrapper
        .find(Component)
        .simulate(Component === Input ? 'change' : 'click', event)

      expect(handleUpdate).toHaveBeenCalledWith(value, getProps().name)
    })

    it('filters the list adnd shows options', () => {
      wrapper
        .find(DropdownItem)
        .forEach((item, index) =>
          expect(item.text()).toBe(getProps().values[index])
        )

      expect(wrapper.find(DropdownItem)).toHaveLength(options)
    })
  })

  it(`matches snapshot`, () => {
    const wrapper = shallow(<TestedComponent {...getProps()} />)

    expect(wrapper).toMatchSnapshot()
  })

  it(`shows valid status`, () => {
    const wrapper = shallow(<TestedComponent {...getProps({ valid: true })} />)

    expect(
      wrapper
        .find(Input)
        .dive()
        .prop('status')
    ).toBe('success')

    expect(wrapper.find(CoinEarned)).toHaveLength(1)
  })
})
