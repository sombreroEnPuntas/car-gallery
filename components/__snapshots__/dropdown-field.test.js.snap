// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownField matches snapshot 1`] = `
<Fragment>
  <FormLineWrap>
    <Input
      disabled={false}
      handleUpdate={
        [MockFunction] {
          "calls": Array [
            Array [
              "r",
              "dinosaur-name",
            ],
            Array [
              "x",
              "dinosaur-name",
            ],
            Array [
              "t-rex",
              "dinosaur-name",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      name="dinosaur-name"
      onChange={[Function]}
      status={null}
      type="text"
      value=""
    />
  </FormLineWrap>
</Fragment>
`;
